### DISCUSSION TITLE
YaSolR MQTT Values Sent

### DISCUSSION THREAD

**Discussion by hallard** [mathieucarbou/YaSolR#36] [General & Support]
Hi,
may be stupid question, but once setting MQTT send for example every 5 seconds, does the values sent are the latest read, or average in between? I'm asking because I tried Remote JSY and sees values sometimes varies a lot in a quick time interval so I'm wondering how it's handled in YaSolR

**Comment by mathieucarbou** [comment]
Mqtt values are refreshed each 5 sec: this can be moved to a larger amount with a slider but cannot be less 
Mqtt publishing is an extensive I/o task, that's why.

**Reply by mathieucarbou** [reply]
MycilaJSY updates it's dashboard each second or 3 times per second I don't remember compared. 
But this is local.
YaSolR internally (and on its dashboard) refreshes faster. Internally this is as fast as possible but this is not possible for example to have this speed of refresh over mqtt

**Reply by hallard** [reply]
Sure, got it, but my question was more about the value sent, lasted read or average of read in between?

FYI I'm updating MQTT every seconds with great success, I must admit that I don't know if this impact routing because I'm not using this feature.

**Reply by mathieucarbou** [reply]
> Sure, got it, but my question was more about the value sent, lasted read or average of read in between?

Oh sorry! All last reads yes!

> FYI I'm updating MQTT every seconds with great success

In yasolr ? The slider does not allow that... Except if you have cheated and went through the backdoor config page 🤣

**Reply by mathieucarbou** [reply]
Note: If you activate debug mode in yasolr and go to the web console, you will see stats about memory and task execution time. You can see the time taken by MQTT publish task in a sort of histogram bin.

These duration vary greatly depending on WiFi / Eth / WiFi signal / Mqtt server / mqqts enabeld or not / etc.

If you have very low mqtt publish times, let me know... 

But I can decrease the slider to 1 and set a default to 5.

**Reply by hallard** [reply]
> In yasolr ? The slider does not allow that... Except if you have cheated and went through the backdoor config page 🤣

yes config page :-)
Not really a backdoor from my point of view because when trying YaSolR (the free version not pro) I was unable to get left menu (makes senses because these are called tab on dash and only available in pro version) all in square red are not available so the only way to try or configure YaSolR is by `/config` page. It took my brain some time before understanding why I didn't had menu you mention everywhere.

![image](https://github.com/user-attachments/assets/5dda5586-87bf-4087-9c90-deec12f88699)


>  If you activate debug mode in yasolr and go to the web console, you will see stats about memory and task execution time. You can see the time taken by MQTT publish task in a sort of histogram bin

Awesome, good to know 

> These duration vary greatly depending on WiFi / Eth / WiFi signal / Mqtt server / mqqts enabeld or not / etc.

I'm using 1s WiFi MQTT but not MQTTs, no Output activated (so no real routing), I need to try Ethernet but got only Olimex POE (not supported). BTW is this because Network Chip is different than ETH-01? 

> But I can decrease the slider to 1 and set a default to 5.

5s default publish is fine, letting 1s for advanced users from config makes sense.

**Reply by mathieucarbou** [reply]
> Olimex POE (not supported). BTW is this because Network Chip is different than ETH-01?

I think they will work.  The network lib I did (MycilaESPConnnect) supports any ETH board on both Arduino Core 2 and 3, for SPI and non-SPI modules.

The problem is that I do not have one at home so I am not able to test, provide a pin mapping for the ETH support and a pin mapping for yasolr.

There are akso several boards: esp32-poe, esp32-poe-iso, esp32-gateway, esp32-evb... 

But if you look at the pio file in the project, there is a section for esp32_poe which is commented. You could if you want test with the OSS version. 

Let me know if you are interested to help me on that, I am willing to add profiles for those, I just cannot test them and considering their price and the variety of board in pio for Olimex, I won't proactively add support for them.

**Reply by hallard** [reply]
Yeah got some ESP32 POE olimex and ESP32 Gateway and even [thingpulse gateway](https://thingpulse.com/product/espgateway-ethernet-esp32-wifi-ble-gateway-with-rj45-ethernet-connector/) so I can do some tests.

Ohh BTW got the license for ESP-DASH Pro also, I tried to build Remote JSY but didn't want to compile even putting the .ino file in src folder and ESP-DASH Pro into local lib folder and changing the Include of course, may be better chance with YaSolR

**Reply by mathieucarbou** [reply]
Yes RemoteJSY depends on the oss version. 
But the pro is compatible also: you just need to change the include if I remember. 
By creating a lib/ESPDASHPro folder in the Sender app, it should work: pio discovers what's in the lib folder by default.

For Yasolr, by tweaking the pio file yes you should be able to use it to build the pro version! Just need to put it in the lib/ESPDASHPro :-)

For WebSerial, there's not a lot of diff between MycilaWebSerial (in my repo) and WebSerial-Pro, except the UI and a few nice things in the UI. I have contributed half the code in his Pro version for the c++ part (a high performance mode to log at a higher rate). So you can stick with MycilaWebSerial.

**Reply by hallard** [reply]
> For Yasolr, by tweaking the pio file yes you should be able to use it to build the pro version! Just need to put it in the lib/ESPDASHPro :-)

That is exactly what I tried and I have the same error than when building RemoteJSY (consistent)

```
Compiling .pio/build/oss-esp32/libd55/MycilaAppInfo/MycilaAppInfo.cpp.o
Compiling .pio/build/oss-esp32/libe0a/MycilaDimmer/MycilaDimmer.cpp.o
Archiving .pio/build/oss-esp32/lib688/libESP-DASH.a
Indexing .pio/build/oss-esp32/lib688/libESP-DASH.a
lib/ESPDASHPro/src/ESPDashPro.cpp: In lambda function:
lib/ESPDASHPro/src/ESPDashPro.cpp:208:3: error: control reaches end of non-void function [-Werror=return-type]
  208 |   });
      |   ^
Compiling .pio/build/oss-esp32/lib136/MycilaRouter/MycilaRouter.cpp.o
Compiling .pio/build/oss-esp32/lib136/MycilaRouter/MycilaRouterOutput.cpp.o
Compiling .pio/build/oss-esp32/lib136/MycilaRouter/MycilaRouterRelay.cpp.o

```
 
![image](https://github.com/user-attachments/assets/e06c3fa9-518d-432b-87ab-353ab972ebaa)



I'm wondering if I don't have a glitch in the source file of dash pro, which version of Dash Pro you're using? I got the 4.6.0

**Reply by hallard** [reply]
Even hardcoding to ignore ESPDASHPro does not seems to works, amazing

![image](https://github.com/user-attachments/assets/50b85619-3aa8-4c2d-81ab-5ea8ffb03849)

**Reply by mathieucarbou** [reply]
> That is exactly what I tried and I have the same error than when building RemoteJSY (consistent)

Yes it is known. There is a bug in the oss and pro version. I have sent a pr weeks ago but he merged just a few minutes ago.

You can have a look at the oss repo - this is the same issue on the same line (return statement on a void function)

**Reply by mathieucarbou** [reply]
Also, it is possible that the ESP-DASH pro version you have is not enough because there are a lot of other code changes that I did in my pro version that Ayush did not pick up (maybe later). One of them is the use of message pack to improve the json serialized size over the Network and use websocket in binary Format instead of text. This allows to reside the required memory by 30-50% and also improves the speed.

There are a few changes like that also.
